### PR TITLE
feat(http-adapter): add new http-adapter

### PR DESCRIPTION
**Pull request by tdeekens**
#### Summary

A more generic http adapter.

### CODE REVIEW

File: packages/graphql-adapter/src/adapter/adapter.ts [hunk 1]
```diff
@@ -8,8 +8,8 @@ import type {
   TFlags,
   TFlagsUpdateFunction,
   TFlagsChange,
-  TGraphQLAdapterInterface,
-  TGraphQLAdapterArgs,
+  TGraphQlAdapterInterface,
+  TGraphQlAdapterArgs,
```
Comment (tdeekens) [inline packages/graphql-adapter/src/adapter/adapter.ts:12]: Unrelated.

**Review by tdeekens — COMMENTED** [review]
This @emmenko is a very simple `http-adapter`. If you have time please have a look. Start the types. Go to the tests then to the code.

**Comment by tdeekens** [comment]
This has been published from my machine as `0.0.1` so we can try it out already.

File: packages/http-adapter/src/adapter/adapter.spec.js [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+      it('should invoke and trigger `warning`', () => {
+        expect(warning).toHaveBeenCalledWith(false, expect.any(String));
+      });
+    });
+  });
+});
+
+describe('when configured', () => {
+  let adapterArgs = {
+    execute: jest.fn().mockResolvedValue({ enabled: true, disabled: false }),
```
Comment (emmenko) [inline packages/http-adapter/src/adapter/adapter.spec.js:44]: Should you return a Promise here?

**Review by emmenko — COMMENTED** [review]
That was quick 😁

Had a look, generally looks good and solid 👌
